rdflib/plugins/sparql/evaluate.py

```diff
@@ -214,7 +214,10 @@ def evalPart(ctx, part):
             pass  # the given custome-function did not handle this part
 
     if part.name == 'BGP':
-        return evalBGP(ctx, part.triples)  # NOTE pass part.triples, not part!
+        # Reorder triples patterns based on 'bindedness' based on current ctx
+        triples = sorted(part.triples, key=lambda t: len([n for n in t if ctx[n] is None]))
```

Comment:
i have problems understanding this... could you maybe expand the comment a bit more?

my guess is `n` is a node in triple `t`, so if `ctx[n] is None` then `n` is unbound, so you select those triples first which have least unbound nodes?